igor-web/src/main/java/com/netflix/spinnaker/igor/helm/HelmMonitor.java

```diff
@@ -0,0 +1,207 @@
+  }
+
+  @Override
+  protected HelmPollingDelta generateDelta(PollContext ctx) {
+    final String account = ctx.partitionName;
+    log.info("Checking for new Helm Charts {}", kv("account", account));
+
+    List<HelmDelta> deltas = new ArrayList<>();
+
+    HelmIndex index = helmAccounts.getIndex(account);
+    if (index == null) {
+      log.error("Failed to fetch index {}", kv("account", account));
+    } else {
+      Set<String> cachedCharts = cache.getCharts(account);
+
+      // If we have no cache at all, do not fire any trigger events
+      // This is so we don't trigger every chart version if Redis dies
+      boolean eventable = !cachedCharts.isEmpty();
+
+      index.entries.forEach(
+          (key, charts) ->
+              charts.forEach(
+                  chartEntry -> {
+                    if (!StringUtils.isEmpty(chartEntry.digest)
+                        && !cachedCharts.contains(chartEntry.digest)) {
+                      deltas.add(
+                          new HelmDelta(
+                              chartEntry.name, chartEntry.version, chartEntry.digest, eventable));
+                    }
+                  }));
+    }
+
+    if (!deltas.isEmpty()) {
+      List<HelmDelta> eventableDeltas =
+          deltas.stream().filter(it -> it.eventable).collect(Collectors.toList());
+      log.info("Found {} new Helm charts {}", eventableDeltas.size(), kv("account", account));
+    }
+
+    return new HelmPollingDelta(account, deltas);
+  }
+
+  @Override
+  protected void commitDelta(HelmPollingDelta deltas, boolean sendEvents) {
+    List<String> digests = deltas.items.stream().map(i -> i.digest).collect(Collectors.toList());
+
+    // Cache results
+    cache.cacheChartDigests(deltas.account, digests);
+
+    // Send events, if needed
+    deltas.items.forEach(
+        item -> {
+          if (sendEvents && item.eventable) {
+            sendEvent(deltas.account, item);
+          }
+        });
+
+    log.info(
+        "Last Helm poll took {} ms {}",
+        System.currentTimeMillis() - deltas.startTime,
+        kv("account", deltas.account));
+  }
+
+  @Override
+  public String getName() {
+    return "helmMonitor";
+  }
+
+  private void sendEvent(String account, HelmDelta delta) {
+    if (!echoService.isPresent()) {
+      log.warn("Cannot send Helm notification: Echo is not enabled");
+      registry
+          .counter(missedNotificationId.withTag("monitor", getClass().getSimpleName()))
+          .increment();
+      return;
+    }
+
+    log.info(
+        "Sending trigger event for {}:{} {}", delta.name, delta.version, kv("account", account));
+    GenericArtifact helmArtifact =
+        new GenericArtifact("helm/chart", delta.name, delta.version, account);
+
+    HelmEvent.Content helmContent =
+        new HelmEvent.Content(account, delta.name, delta.version, delta.digest);
+
+    echoService.get().postEvent(new HelmEvent(helmContent, helmArtifact));
```

Comment:
I think you need to wrap this request in `AuthenticatedRequest.allowAnonymous` like [here](https://github.com/spinnaker/igor/blob/7976bd460d2fa61442cb6e2f75e662be4c87e263/igor-web/src/main/groovy/com/netflix/spinnaker/igor/docker/DockerMonitor.groovy#L183)